File: python/leetcode121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left=0
        right=left+1
        

        profit=0

        while right<len(prices):
            if prices[left]>prices[right] or prices[left]==prices[right]:
                left=right
                right+=1
            elif prices[left]<prices[right]:
                profit=max(profit,prices[right]-prices[left])
                right+=1

        return profit

File: java/leetcode121.java
class Solution {
    public int maxProfit(int[] prices) {
        int left=0,right=1,profit=0;

        while(right<prices.length){
            if(prices[left]<prices[right]){
                profit= Math.max(profit,prices[right]-prices[left]);
                right++;
            }else{
                left=right;
                right++;
            }
        }

        return profit;

    }
}